core/arch/arch.h

```diff
@@ -117,7 +119,9 @@ mixed_mode_enabled(void)
 #    define SCRATCH_REG5_OFFS XDI_OFFSET
 #elif defined(AARCHXX)
 #    define R0_OFFSET ((MC_OFFS) + (offsetof(priv_mcontext_t, r0)))
+#    define REG0_OFFSET R0_OFFSET
```

Comment:
I liked the IF_X86_ELSE(XAX_OFFSET, R0_OFFSET) as it was before more, maybe just a preference. I thought it was more readable.

Reply:
Really?  The offset had zero relation to the register, though: why did you want to see the split by architecture?  The rseq code just wants two TLS slots.  It doesn't care what register is typically put there by the stubs and ibl gencode.

Reply:
I.e., the rseq code is storing a random register into the first slot.  I would think it would be more confusing to see it writing the random register to "the xax slot for x86, or the r0 slot for arm".  Wouldn't you rather see "the first slot"?

Reply:
Ok I see what you're saying. Ok then!